autonomy/cli/scaffold_fsm_templates/tests.py

```diff
@@ -0,0 +1,324 @@
+                    synchronized_data_update_fn=lambda sync_data, _: sync_data.update(**test_case.final_data),
+                    synchronized_data_attr_checks=test_case.synchronized_data_attr_checks,
+                    exit_event=test_case.event,
+                    **kwargs,  # varies per BaseRoundTestClass child
+                )
+            )
+
+    """
+
+    TEST_ROUND_CLS = """\
+    class Test{RoundCls}(Base{FSMName}RoundTest):
+        \"\"\"Tests for {RoundCls}.\"\"\"
+
+        round_class = {RoundCls}
+
+        # TODO: provide test cases
+        @pytest.mark.parametrize("test_case, kwargs", [])
+        def test_run(self, test_case: RoundTestCase, **kwargs: Any) -> None:
+            \"\"\"Run tests.\"\"\"
+
+            self.run_test(test_case, **kwargs)
+
+    """
+
+
+@dataclass
+class TEST_BEHAVIOURS:
+    """Template for test_behaviours.py"""
+
+    FILENAME = "test_behaviours.py"
+
+    HEADER = """\
+    \"\"\"This package contains round behaviours of {AbciApp}.\"\"\"
+
+    from pathlib import Path
+    from typing import Any, Dict, Hashable, Optional, Type
+    from dataclasses import dataclass
+
+    import pytest
+
+    from packages.valory.skills.abstract_round_abci.base import AbciAppDB
+    from packages.valory.skills.abstract_round_abci.behaviours import (
+        AbstractRoundBehaviour,
+        BaseBehaviour,
+        make_degenerate_behaviour,
+    )
+    from packages.{author}.skills.{skill_name}.behaviours import (
+        {FSMName}BaseBehaviour,
+        {FSMName}RoundBehaviour,
+        {behaviours},
+    )
+    from packages.{author}.skills.{skill_name}.rounds import (
+        SynchronizedData,
+        DegenerateRound,
+        Event,
+        {AbciApp},
+        {all_rounds},
+    )
+
+    from packages.valory.skills.abstract_round_abci.test_tools.base import (
+        FSMBehaviourBaseCase,
+    )
+
+
+    @dataclass
+    class BehaviourTestCase:
+        \"\"\"BehaviourTestCase\"\"\"
+
+        initial_data: Dict[str, Hashable]
+        event: Event
+
+    """
+
+    BASE_BEHAVIOUR_TEST_CLS = """\
+    class Base{FSMName}Test(FSMBehaviourBaseCase):
+        \"\"\"Base test case.\"\"\"
+
+        path_to_skill = Path(__file__).parent.parent
+
+        behaviour: {FSMName}RoundBehaviour
+        behaviour_class: Type[{FSMName}BaseBehaviour]
+        next_behaviour_class: Type[{FSMName}BaseBehaviour]
+        synchronized_data: SynchronizedData
+        done_event = Event.DONE
+
+        def fast_forward(self, data: Optional[Dict[str, Any]] = None) -> None:
+            \"\"\"Fast-forward on initialization\"\"\"
+
+            data = data if data is not None else {{}}
+            self.fast_forward_to_behaviour(
+                self.behaviour,
+                self.behaviour_class.behaviour_id,
+                SynchronizedData(AbciAppDB(setup_data=AbciAppDB.data_to_lists(data))),
+            )
+            assert self.behaviour.behaviour_id == self.behaviour_class.behaviour_id
```

Comment:
As discussed this should be `self.behaviour.current_behaviour.behaviour_id` instead of `self.behaviour.behaviour_id`.